Source/Applications/openXDA/openXDA/Controllers/WebAPI/WebAPI.cs

```diff
@@ -392,6 +399,28 @@ private string getEventCharacteristicFilter(EventPost postData)
             return string.Join(" AND ", characteristics);
         }
 
+        private string getTimeExclusionFilter(EventPost postData)
+        {
+            IEnumerable<int> hours = postData.TimeFilters.Where(filt => filt < TimeFilter.Sunday).Select(filt => (int) filt);
+            // SQL dw is 1-7, starting on sunday
+            IEnumerable<int> days = postData.TimeFilters.Where(filt => filt >= TimeFilter.Sunday && filt < TimeFilter.Week00).Select(filt => filt - TimeFilter.Sunday + 1);
+            // Influx uses the iso week standard, i.e. 1-53
+            IEnumerable<int> weeks = postData.TimeFilters.Where(filt => filt >= TimeFilter.Week00 && filt < TimeFilter.January).Select(filt => filt - TimeFilter.Week00 + 1);
+            IEnumerable<int> months = postData.TimeFilters.Where(filt => filt >= TimeFilter.January).Select(filt => filt - TimeFilter.January + 1);
+
+            string hourFilt = hours.Count() != 0 ? $"DATEPART(hh,Event.StartTime) not in ({string.Join(",", hours)})" : null;
+            string dayFilt = days.Count() != 0 ? $"DATEPART(dw, Event.StartTime) not in ({string.Join(",", days)})" : null;
+            string weekFilt = weeks.Count() != 0 ? $"DATEPART(isowk, Event.StartTime) not in ({string.Join(",", weeks)})" : null;
+            string monthFilt = months.Count() != 0 ? $"DATEPART(mm, Event.StartTime) not in ({string.Join(",", months)})" : null;
+
+            string filters =
+                $"{(string.IsNullOrEmpty(hourFilt) ? "" : $"AND {hourFilt}")}" +
+                $"{(string.IsNullOrEmpty(dayFilt) ? "" : $"AND {dayFilt}")}" +
+                $"{(string.IsNullOrEmpty(weekFilt) ? "" : $"AND {weekFilt}")}" +
+                $"{(string.IsNullOrEmpty(monthFilt) ? "" : $"AND {monthFilt}")}";
```

Comment:
Maybe it would be better to build the filter string from an enumerable.
```c#
string[] filters = { hourFilt, dayFilt, weekFilt, monthFilt };
return string.Join(" AND ", filters.Where(f => !(f is null)));
```

Reply:
Good idea, added